client/src/app/components/audit-template/audit-template-view/audit-template-view.component.ts

```diff
@@ -67,6 +129,83 @@ export abstract class AuditTemplateViewComponent implements OnInit {
     }
   }
 
+  timeZoneChange(event: { value: { utc: string[]; }; }): void {
+    this.timeZoneUTC = event.value.utc[0];
+  }
+
+  recurrenceExpand(): void {
+    this.repeatEvery = '1';
+    this.isRecurrenceChosen = true;
+    this.panelOpenState = true;
+  }
+
+  recurrenceCollapsed(): void {
+    this.isRecurrenceChosen = false;
+    this.panelOpenState = false;
+    this.allDaysChecked = false;
+    this.allMonthsChecked = false;
+    this.repeatEvery = '1';
+    this.recurrenceDay.subCheckBox?.forEach((t) => (t.checked = false));
+    this.recurrenceMonth.subCheckBox?.forEach((t) => (t.checked = false));
+    // @ts-ignore
+    this.recurrenceDay.subCheckBox[0].checked = true;
+    // @ts-ignore
+    this.recurrenceMonth.subCheckBox[0].checked = true;
+    this.errorMessageCheckboxDay = ' ';
+    this.errorMessageCheckboxMonth = ' ';
+  }
+
+  updateAllCheckbox(type: string): void {
+    if (type === 'dayCheckbox') {
+      this.allDaysChecked =
+        this.recurrenceDay.subCheckBox != null &&
+        this.recurrenceDay.subCheckBox.every((t) => t.checked);
+      this.errorMessageCheckboxDay = ' ';
+    } else if (type === 'monthCheckbox') {
+      this.allMonthsChecked =
+        this.recurrenceMonth.subCheckBox != null &&
+        this.recurrenceMonth.subCheckBox.every((t) => t.checked);
+      this.errorMessageCheckboxMonth = ' ';
+    }
+  }
+
+  // @ts-ignore
+  someCheckbox(type: string): boolean {
+    if (type === 'dayCheckbox') {
+      if (this.recurrenceDay.subCheckBox == null) {
+        return false;
+      }
+      return (
+        this.recurrenceDay.subCheckBox.filter((t) => t.checked).length > 0 &&
+        !this.allDaysChecked
+      );
+    } else if (type === 'monthCheckbox') {
+      if (this.recurrenceMonth.subCheckBox == null) {
+        return false;
+      }
+      return (
+        this.recurrenceMonth.subCheckBox.filter((t) => t.checked).length > 0 &&
+        !this.allMonthsChecked
+      );
+    }
+  }
```

Comment:
Same applies here, I'm not sure if this would fit with the HTML, but if yes, moving this into 2 functions would make more sense to avoid having to check.

Reply:
someCheckbox has now been split into two function named: someCheckboxDay and someCheckboxMonth.